baseapp/abci_utils.go

```diff
@@ -424,7 +424,9 @@ func (ts *defaultTxSelector) Clear() {
 }
 
 func (ts *defaultTxSelector) SelectTxForProposal(_ context.Context, maxTxBytes, maxBlockGas uint64, memTx sdk.Tx, txBz []byte) bool {
-	txSize := uint64(cmttypes.ComputeProtoSizeForTxs([]cmttypes.Tx{txBz}))
+	// compute the total size of the selected transactions + the prospective transaction
+	allTxs := append(ts.selectedTxs, txBz)
```

Comment:
I'm afraid this might modify `ts.selectedTxs` in some cases.

Reply:
Actually, nvm. I think this is probably fine. But to be safe, I would copy the slice and then append.

Reply:
Did some googleing, looks like `append` is safe to use here as it performs a deep copy, just like `copy`. If this is not what you are referring to lmk